### PR TITLE
Kashirin Alexander Lab 2 Var 4

### PR DESCRIPTION
**Pull request by AlexanderVladislavovich**

### CODE REVIEW

File: llvm/labs/lab2/kashirin_alexander/BitShiftPass.cpp [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        llvm::BinaryOperator *op = llvm::cast<llvm::BinaryOperator>(&Inst);
+        if (op->getOpcode() != llvm::Instruction::BinaryOps::Mul) {
+          continue;
+        }
+
+        llvm::Value *leftOper = op->getOperand(0);
+        llvm::Value *rightOper = op->getOperand(1);
+
+        int logVal1 = getLogBase2(leftOper);
+        int logVal2 = getLogBase2(rightOper);
```
Comment (aobolensk) [inline llvm/labs/lab2/kashirin_alexander/BitShiftPass.cpp]: What do `logVar1` and `logVar2` mean?

File: llvm/labs/lab2/kashirin_alexander/BitShiftPass.cpp [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        if (auto *SI = llvm::dyn_cast<llvm::StoreInst>(Inst)) {
+          StInst = SI;
+        }
+        if (Inst == LI) {
+          break;
+        }
+      }
+      Op->reverseUseList();
+      if (!StInst) {
+        return -2;
```
Comment (aobolensk) [inline llvm/labs/lab2/kashirin_alexander/BitShiftPass.cpp]: Why -2 ?

File: llvm/test/lab2/kashirin_alexander/BitShiftPass_test.ll [hunk 3]
```diff
@@ -0,0 +1,214 @@
+; RUN: split-file %s %t
```
Comment (aobolensk) [inline llvm/test/lab2/kashirin_alexander/BitShiftPass_test.ll]: Come up with **your own** tests

**Review by aobolensk — CHANGES_REQUESTED** [review]

File: llvm/test/lab2/kashirin_alexander/BitShiftPass_test.ll (hunk 3 above)
Comment (AlexanderVladislavovich) [reply llvm/test/lab2/kashirin_alexander/BitShiftPass_test.ll]: tests updated

File: llvm/labs/lab2/kashirin_alexander/BitShiftPass.cpp (hunk 1 above)
Comment (AlexanderVladislavovich) [reply llvm/labs/lab2/kashirin_alexander/BitShiftPass.cpp]: logVar1 is a base 2 logarithm of left operand, logVar2 is a base 2 logarithm of right operand

**Review by aobolensk — APPROVED** [review]

**Pull request merged**